app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -139,10 +115,10 @@ class WallpaperManager(
     /**
      * Load a wallpaper that is saved locally.
      */
-    fun loadSavedWallpaper(context: Context, wallpaper: Wallpaper): Bitmap? =
-        when (wallpaper) {
-            is Wallpaper.Local -> loadWallpaperFromDrawables(context, wallpaper)
-            is Wallpaper.Remote -> loadWallpaperFromDisk(context, wallpaper)
+    fun Wallpaper.load(context: Context): Bitmap? =
```

Comment:
nit: Should this extension instead live in a extension function file?

Reply:
This one technically could, but structuring it this way was an intentional plan for refactoring the `WallpaperManager` as a `WallpaperUseCase`. It forces a client to have a reference to the `UseCase` to be able to load wallpapers. That way, we can treat the UseCase as an abstraction barrier. Right now, all the loading is done using implicit `Context`-dependent stuff, but I imagine the callstack would look more like this soon:

```
with (wallpaperUseCase) {
    wallpaper.load() -> wallpaperUseCase.fileManager.loadWallpaper()
}
```

Not sure that notation really makes sense lol. But basically we could inject the `context` (or a `WallpaperLoader` or whatever else we wanted to use to actually do disk reading) into the object hierarchy and hide it from call sites of `wallpaper.load()`.

Sidenote: the pattern here might be more clear if we were on Kotlin 1.6.20 and could use [context receivers](https://blog.jetbrains.com/kotlin/2022/02/kotlin-1-6-20-m1-released/#prototype-of-context-receivers-for-kotlin-jvm).